readme.md

```diff
@@ -9,13 +11,56 @@
+
+### Signing Requests
+
+Given the following POST parameters
+```
+{
+    "market": "REP/WETH"
+    "state": "all"
+    "nonce": 1234567
+}
+```
+
+We create a payload message by ordering the payload by the keys then concatinating the keys together followed by the values for those keys concatinated together. So for the above payload we would get the following payload message
+
+```
+marketnoncestateREP/WETH1234567all
+```
+
+This message is then hashed using Keccak-256 and signed by the private key for the ethereum account to produce a vrs signature which we include in the header of the request.
+
+
+## Deposits/Withdrawals
+
+Paradex is a non custodial decentralised trading platform. This means you retain complete control of your funds and they stay within your own wallet until a trade takes place. As such there is no way to deposit or withdraw funds to or from Paradex. Assets can be moved in and out of your wallet using the normal means at your disposal either programmatically or via an application. What you will need to do though is wrap Eth and set allowances. Please see below
+
+## Wrapping ETH and Setting Allowances
+
+Paradex enables the decentralised trading of ERC20 tokens. The ERC20 standard was created to provide a common interface for how tokens will function and was created after the initial Ethereum standard. As such Ethereum is not currently an ERC20 token and has to be converted to a compatible ERC20 Ethereum called Wrapped-ETH (WETH). Unsurprisingly this conversion process is called wrapping with 1 WETH being equivalent to 1 ETH in value.
+Another feature of ERC20 tokens is allowances. Allowances allow you to control how much if any of your funds can be transfered by out of your wallet by Paradex. By default your allowances are set to 0 so before you begin trading you have to set allowances for your tokens. Remember even once you have set your allowances no funds can be transferred out of your account without you putting a valid order on the orderbook. If you try and place an order to paradex without setting your allowances for that token your order will enter an unfunded state. If you are wanting to set allowances programmatically the 0x.js library provides some convenience methods to help you do that https://www.0xproject.com/docs/0xjs#token
```

Comment:
Minor quibble, but it isn't actually Paradex who gets the allowance set and does the transfers, it's the 0x smart contract.

Reply:
Capitalize Paradex

Reply:
change to:
If you want to set allowances programmatically, the 0x.js library provides some convenience methods to help you do that: https://www.0xproject.com/docs/0xjs#token